cdap-docs/developers-manual/source/data-exploration/tables.rst

```diff
@@ -0,0 +1,107 @@
+Table Exploration
+============================================
+
+A ``Table`` is a core Dataset. Unlike relational database tables where every
+row has the same schema, every row of a Table can have a different set of columns.
+Though Tables do not require a schema, in practice they are often written with an
+implicit schema. Column names are often strings, with a single data type used
+for all values in the same column. If you are using a Table in this way,
+you can set the schema Table property to enable exploration. The schema will be
+applied at read time, allowing you to run ad-hoc queries against the Table. 
+
+Requirements
+------------
+In order to explore a Table, your Table must meet a few requirements.
+
+* Columns names must be strings.
+
+* All column values for a specific column must be of the same type. For example, a value cannot be a string
+  in one row and an integer in another.
+
+* Column values must be of a primitive type.
+  A primitive type is one of boolean, int, long, float, double, bytes, or string. 
+
+* Column names must be valid Hive column names. This means they cannot be reserved keywords like 'drop'.
+  Please refer to the `Hive language manual <https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DDL>`__
+  for more information about Hive.
+
+Creating an Explorable Table
+----------------------------
+
+When creating a ``Table`` in your application, if you set the schema table property, your Table
+will be enabled for exploration after it is created::
+
+  @Override
+  public void configure() {
+    Schema profileSchema = Schema.recordOf(
+      "profile",
+      // id, name, and email are never null and are set when a user profile is created
+      Schema.Field.of("id", Schema.of(Schema.Type.STRING)),
+      Schema.Field.of("name", Schema.of(Schema.Type.STRING)),
+      Schema.Field.of("email", Schema.of(Schema.Type.STRING)),
+      // login and active are never set when a profile is created but are set later, so they are nullable.
+      Schema.Field.of("login", Schema.nullableOf(Schema.of(Schema.Type.LONG))),
+      Schema.Field.of("active", Schema.nullableOf(Schema.of(Schema.Type.LONG)))
+    );
+    createDataset("profiles", Table.class.getName(), DatasetProperties.builder()
+      // set the schema property so that it can be explored via Hive
+      .add(Table.PROPERTY_SCHEMA, profileSchema.toString())
+      // to indicate that the id field should come from the row key and not a row column
+      .add(Table.PROPERTY_SCHEMA_ROW_FIELD, "id")
+      .build());
+  } 
+
+Note that the schema row field property is set along with the schema property. The schema row field property
+must be set if you want to explore your Table row along with Table columns. In the example above, this property
+will let CDAP know to read the ``id`` field from the Table row instead of from the Table columns. 
+
+Setting Schema on an Existing Table
+-----------------------------------
+
```

Comment:
To display the curl call correctly, insert this formatting hint on the next line:
```
.. highlight:: console
```